### PR TITLE
Ensure components objects deletion and creation at e2e tests

### PR DESCRIPTION
**Pull request by ormergi**
This PR add checks in e2e tests to ensure deletion and deployment of Secrets and MutationWebhookConfiguration objects

### CODE REVIEW

**Comment by ormergi** [comment]
> looks good. Let's also add a checkForSecret() function in [checkForComponent()](https://github.com/kubevirt/cluster-network-addons-operator/blob/dfe19b2ff50d68e90f98a598fef0f82b555988c1/test/check/check.go#L229)

Hey thank for reviewing, my intent is to extend this to also handle webhookConfiguration objects

why do we need to add this kind of function?

File: test/e2e/workflow/deployment_test.go [hunk 1]
```diff
@@ -165,6 +165,7 @@ var _ = Describe("NetworkAddonsConfig", func() {
 		//2300
 		It("should be able to remove the config and create it again", func() {
 			DeleteConfig()
+			CheckComponentsRemoval(components)
```
Comment (phoracek) [inline test/e2e/workflow/deployment_test.go:168]: This is adding a regression to our tests. Please document it with TODO and a linked issue. We need to drop this once we resolve it.

File: test/check/check.go [hunk 2]
```diff
@@ -377,6 +381,11 @@ func checkForSecurityContextConstraintsRemoval(name string) error {
 	return isNotFound("SecurityContextConstraints", name, err)
 }
 
+func checkForSecretsRemoval(name string) error {
+		err := framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name, Namespace: components.Namespace}, &corev1.Secret{})
```
Comment (phoracek) [inline test/check/check.go:385]: Please indent it with a single tab.

**Comment by RamLavi** [comment]
>why do we need to add this kind of function?

to make sure the component (i.e. secret)  is there once the cluster is ready

File: test/check/check.go [hunk 3]
```diff
@@ -377,6 +402,16 @@ func checkForSecurityContextConstraintsRemoval(name string) error {
 	return isNotFound("SecurityContextConstraints", name, err)
 }
 
+func checkForSecretRemoval(name string) error {
+	err := checkForSecret(name)
```
Comment (phoracek) [inline test/check/check.go:406]: I like the idea, but I hate partial re-factoring. Please open a PR where you adjust all `*Removal` functions to this format.

File: test/e2e/workflow/deployment_test.go (hunk 1 above)
Comment (ormergi) [reply test/e2e/workflow/deployment_test.go:170]: done

File: test/check/check.go (hunk 3 above)
Comment (ormergi) [reply test/check/check.go:406]: ack

File: test/check/check.go [hunk 4]
```diff
@@ -359,6 +376,14 @@ func checkForDaemonSet(name string) error {
 	return nil
 }
 
+func checkForSecret(name string) error {
+	return framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name}, &corev1.Secret{})
```
Comment (phoracek) [inline test/check/check.go:380]: As Quique commented offline, this and the following function misses namespace definition.

File: test/check/check.go [hunk 5]
```diff
@@ -377,6 +402,16 @@ func checkForSecurityContextConstraintsRemoval(name string) error {
 	return isNotFound("SecurityContextConstraints", name, err)
 }
 
+func checkForSecretRemoval(name string) error {
+	err := framework.Global.Client.Get(context.Background(), types.NamespacedName{Name: name,Namespace: components.Namespace}, &corev1.Secret{})
```
Comment (qinqon) [inline test/check/check.go:406]: You can use `checkForSecret` here.

Comment (ormergi) [reply test/check/check.go:406]: I will change all `checkFor*Removal` function to use `checkFor*` in another PR

File: test/check/check.go (hunk 4 above)
Comment (ormergi) [reply test/check/check.go]: done

File: test/check/check.go (hunk 5 above)
Comment (qinqon) [reply test/check/check.go:406]: fair enough

**Comment by phoracek** [comment]
The test fail looks related.

**Comment by phoracek** [comment]
@ormergi why do you rerun test that is clearly related to the code you just added?

**Comment by ormergi** [comment]
> @ormergi why do you rerun failed test that is clearly related to the code you just added?

Sorry about that, I misunderstood travis status

**Pull request merged**